From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: hopeless-ponderer <hopelessponderer1123@gmail.com>
Date: Sat, 12 Jun 2021 17:23:40 -0700
Subject: [PATCH] updated TODO.md

Foundation component transfers actor seed to new entity on EntityRenamed

Roman fertility temple renamed to Aedes Saturni (Temple of Saturn)
---
 binaries/data/mods/public/TODO.md                             | 2 --
 binaries/data/mods/public/simulation/components/Foundation.js | 4 ++++
 .../templates/structures/rome/temple_patron_fertility.xml     | 2 +-
 3 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/binaries/data/mods/public/TODO.md b/binaries/data/mods/public/TODO.md
index 6713a735d0..ca34f6f4c4 100644
--- a/binaries/data/mods/public/TODO.md
+++ b/binaries/data/mods/public/TODO.md
@@ -68,8 +68,6 @@ Please feel free to contribute if you see anything you can do here. If there is
 	- Alternatively, give Celts and Iberians rushing bonuses to offset their weaker city-building abilities
 - Add "thrust" damage, for spear units
 - Implement "Charge" attack. One-time, extremely strong attack that a unit generates by running a certain distance. Used for spear cav and ships
-- Create city name manager, implement random city naming
-	- Create gui widget for renaming cities (start with a cheat in the short run)
 - Create dropdown sub-menus for random civ groups, to reduce clutter. Implement by adding "Category" to each civ.json, naming which sub-menu the group will be sorted into.
 - Create unique structure artwork for special temples.
 - Create builder/engineer unit to build structures of city level and higher
diff --git a/binaries/data/mods/public/simulation/components/Foundation.js b/binaries/data/mods/public/simulation/components/Foundation.js
index f567b0120b..e62b31eece 100644
--- a/binaries/data/mods/public/simulation/components/Foundation.js
+++ b/binaries/data/mods/public/simulation/components/Foundation.js
@@ -441,6 +441,10 @@ Foundation.prototype.OnEntityRenamed = function(msg)
 	let cmpFoundationNew = Engine.QueryInterface(msg.newentity, IID_Foundation);
 	if (cmpFoundationNew)
 		cmpFoundationNew.AddBuilders(this.GetBuilders());
+	const cmpVisual = Engine.QueryInterface(this.entity, IID_Visual);
+	const cmpVisualNew = Engine.QueryInterface(msg.newentity, IID_Visual);
+	if (cmpVisual && cmpVisualNew)
+		cmpVisualNew.SetActorSeed(cmpVisual.GetActorSeed());
 };
 
 function FoundationMirage() {}
diff --git a/binaries/data/mods/public/simulation/templates/structures/rome/temple_patron_fertility.xml b/binaries/data/mods/public/simulation/templates/structures/rome/temple_patron_fertility.xml
index 7493781031..4ad1070f9f 100644
--- a/binaries/data/mods/public/simulation/templates/structures/rome/temple_patron_fertility.xml
+++ b/binaries/data/mods/public/simulation/templates/structures/rome/temple_patron_fertility.xml
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Entity parent="civ/rome/temple_patron|template_structure_civic_temple_patron_fertility">
   <Identity>
-    <SpecificName>Aedes Cereris</SpecificName>
+    <SpecificName>Aedes Saturni</SpecificName>
   </Identity>
 </Entity>
-- 
2.25.1

